tools/clang/test/SemaHLSL/hlsl/objects/HitObject/hitobject-entry-errors.hlsl

```diff
@@ -0,0 +1,57 @@
+// RUN: %dxc -T lib_6_9 %s -verify
+
+struct [raypayload] Payload
+{
+    float elem
+          : write(caller,anyhit,closesthit,miss)
+          : read(caller,anyhit,closesthit,miss);
+};
+
+struct Attribs { float2 barys; };
+
+void UseHitObject() {
+// expected-error@+4{{Shader kind 'compute' incompatible with shader execution reordering (has to be raygeneration, closesthit or miss)}}
```

Comment:
Does this test work if you move these expected errors into the corresponding entry functions? If not, would you re-order the entry functions below so that you can move these into the entry functions? Just a design preference, so that it's obvious the error is that it's the call site being in the context of a bad shader kind, not the function definition itself.

Reply:
Moved into entry functions in [57c861a](https://github.com/microsoft/DirectXShaderCompiler/pull/7097/commits/57c861aab39debbbe5ba53dbb2a9bb4a7c40a822).